File: src/programmers/42885/kyoulho.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int answer = 0;

        int left = 0;  // 가장 가벼운 사람의 인덱스
        int right = people.length - 1;  // 가장 무거운 사람의 인덱스

        while (left <= right) {
            // 가장 가벼운 사람과 가장 무거운 사람을 함께 태울 수 있을 경우
            if (people[left] + people[right] <= limit) {
                left++;
            }
            right--;
            answer++;
        }

        return answer;
    }
}

File: src/programmers/42587/kyoulho.kt
import java.util.*

class Solution {
    fun solution(priorities: IntArray, location: Int): Int {
        var answer = 0

        val q: Queue<Pair<Int, Int>> = LinkedList()
        priorities.forEachIndexed { index, priority ->
            q.add(Pair(index, priority))
        }

        while (q.isNotEmpty()) {
            val poll = q.poll()
            if (poll.second < (q.maxByOrNull { it.second }?.second ?: 0)) {
                q.add(poll)
            } else {
                answer++
                if (poll.first == location) break
            }
        }
        return answer
    }

}